### PR TITLE
BaSP-MR-32: SUBSCRIPTION TEST - update and delete

### PR DESCRIPTION
**Pull request by JuanManuel-Blois**
Added the delete and put method tests, for succes code 200 and error code 400

### CODE REVIEW

File: src/tests/subscriptions.test.js [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+import Subscription from '../models/Subscription';
+import subSeed from '../seeds/subscriptions';
+
+beforeAll(async () => {
+  await Subscription.collection.insertMany(subSeed);
+});
+
+describe('PUT /api/subscriptions', () => {
+  test('should update one subscription', async () => {
+    const response = await request(app).put('/api/subscriptions/64694cb404b48dafdecfa44a').send({
```
Comment (nicolasblasco) [inline src/tests/subscriptions.test.js:12]: What happens if we change the subscription ID in the seed file? I think we should capture the id of the first subscription from the seed into a variable. 

We should use this variable throughout our tests.

File: src/tests/subscriptions.test.js [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+      classId: '64694cb404b48dafdecfa12e',
+      members: [
+        '64655cf22c2d037df75e72dd', '64691789d00ee55da969d212',
+      ],
+      date: '2023-10-05T03:00:00.000Z',
+    });
+    expect(response.status).toBe(200);
```
Comment (nicolasblasco) [inline src/tests/subscriptions.test.js:19]: Don't you think we can test the body of the response? We can test the message, data, and error of the body of the response.

We should do this for all of our tests.

File: src/tests/subscriptions.test.js [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+
+describe('PUT /api/subscriptions', () => {
+  test('should update one subscription', async () => {
+    const response = await request(app).put('/api/subscriptions/64694cb404b48dafdecfa44a').send({
+      classId: '64694cb404b48dafdecfa12e',
+      members: [
+        '64655cf22c2d037df75e72dd', '64691789d00ee55da969d212',
+      ],
+      date: '2023-10-05T03:00:00.000Z',
+    });
```
Comment (nicolasblasco) [inline src/tests/subscriptions.test.js:18]: I think we should move this data into a variable called `mockedSubscription` so we can use it thought our tests

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

File: src/tests/subscriptions.test.js [hunk 4]
```diff
@@ -1,5 +1,122 @@
...
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.message).toBe('Subscription Updated Successfully!');
+  });
+
+  test('should send error 400 because this ID doesnt exist', async () => {
+    const response = await request(app).put(`/api/subscriptions/${invalidSubMockup}`).send(modifiedSubscription);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data).toBeDefined();
```
Comment (LucianoAlarcon) [inline src/tests/subscriptions.test.js:47]: Can toy change `.toBeDefined()` to `toEqual({})`

File: src/tests/subscriptions.test.js [hunk 5]
```diff
@@ -1,5 +1,122 @@
...
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.data).toBeDefined();
+    expect(response.body.message).toBe('Subscription Deleted Successfully!');
+  });
+
+  test('should send 400 error because his ID is invalid', async () => {
+    const response = await request(app).delete(`/api/subscriptions/${invalidSubMockup}`).send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data).toBeDefined();
```
Comment (LucianoAlarcon) [inline src/tests/subscriptions.test.js:96]: Can toy change `.toBeDefined()` to `toEqual({})`

**Review by LucianoAlarcon — CHANGES_REQUESTED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**